src/lib_ccaligner/recognize_using_pocketsphinx.cpp

```diff
@@ -455,9 +457,20 @@ bool PocketsphinxAligner::recognise() {
         _rvWord = ps_start_utt(_psWordDecoder);
         _rvWord = ps_process_raw(_psWordDecoder, sample + samplesAlreadyRead, samplesToBeRead, FALSE, FALSE);
         _rvWord = ps_end_utt(_psWordDecoder);
-
         _hypWord = ps_get_hyp(_psWordDecoder, &_scoreWord);
 
+        // If the subtitles' start time is after the audio's finish time, we shifted 
+        // the subtitle's start time back by one second fragments until there was a match 
+        // between the approx and actual subtitles.
+        while (_hypWord == nullptr && samplesAlreadyRead > 16000) {
```

Comment:
This is making a lot of assumptions. Consider the following points/cases:

- _hypWord is going to be null when no utterance is recognised. This may happen when the speech is very unclear or there is lot of noise, or the audio doesn't contain speech at that point at all. You're reducing the start time of recognition to widen the scope - but not changing the length upto which audio is to be recognised (`samplesToBeRead`). This will just shift the recognition window by one second backwards, not sure how it solves the problem.